src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -57,6 +60,53 @@ def handleRequest(request)
         response["X-Puppet-Version"])
   end
 
+  def getClassInfoForAllEnvironments()
```

Comment:
I'm pretty mixed at this point about the work that this PR is doing to try to work around the fact that environment objects and settings can get cached.  My hope is that when this code gets hooked up to other Clojure-level code in Puppet Server later on that that code will always be marking all of the environments associated with a JRubyPuppet instance as "expired".  Granted, we have a separate defect, SERVER-813, where the expiration isn't applied across all instances in some cases.  That defect will need to be fixed at some point, though.  If we assume that the expiration is done before calls to either of the "getClassInfo" methods added by this PR are made, I don't think any of the cache avoidance work would be necessary - including the `clear_environment_settings` call or the `@env_loader.list.find()` call made in `getClassInfoForEnvironment`.  Not knowing how the rest of the dependencies are going to fall out at this point, I went with including the cache avoidance logic for now.  Hopefully / maybe we could rip this out later, though?